web-bundle/src/main/java/com/graphhopper/replica/StreetEdgeExporter.java

```diff
@@ -98,17 +103,34 @@ public List<StreetEdgeExportRecord> generateRecords(EdgeIteratorState iteratorSt
         // Convert GH's distance in meters to millimeters to match R5's implementation
         long distanceMillimeters = distanceMeters * 1000;
 
-        // Fetch OSM ID, skipping edges from PT meta-graph that have no IDs set (getOsmIdForGhEdge returns -1)
-        long osmId = OsmHelper.getOsmIdForGhEdge(ghEdgeId, ghIdToOsmId);
-        if (osmId == -1L) {
+        // Fetch OSM Way ID, skipping edges that have no IDs set (getOSMWay returns -1)
+        long osmWayId = osmHelper.getOSMWay(ghEdgeId);
+        if (osmWayId == -1L) {
+            return output;
+        }
+
+        // Fetch OSM Node IDs for each node of edge
+        long startOsmNode = osmHelper.getOSMNode(osmHelper.getBaseNodeForEdge(ghEdgeId));
+        long endOsmNode = osmHelper.getOSMNode(osmHelper.getNodeAdjacentToEdge(ghEdgeId));
+
+        // Filter out single-point "edges" + edges with identical start/end point locations
+        // and no intermediate points (as would exist in the case of road loops)
+        if (wayGeometry.size() <= 1) {
```

Comment:
Special callout here - I added some extra cases where we filter out edges in the export, mostly based on trial-and-error and finding weird edge cases where we definitely don't want "silly" links included in the export